File: in_go/src/pattern_15.go
package src

import "fmt"

func Pattern_15() {
	row := 9
	column := 5
	var symbol int
	halfRow := row / 2

	for i := 0; i < row; i++ {
		symbol = 1
		for j := 0; j < column; j++ {
			if i <= halfRow {
				if j >= column-1-i {
					fmt.Print(symbol)
					symbol++
				} else {
					fmt.Print(" ")
				}
			} else {
				if j >= i-column+1 {
					fmt.Print(symbol)
					symbol++
				} else {
					fmt.Print(" ")
				}
			}
		}
		fmt.Println()
	}
}

File: in_rust/src/pattern_15.rs
pub fn pattern_15() {
    let row = 9;
    let column = 5;
    let mut symbol;
    let half_row = row / 2;

    for i in 0..row {
        symbol = 1;
        for j in 0..column {
            if i <= half_row {
                if j >= column - 1 - i {
                    print!("{}", symbol);
                    symbol += 1;
                } else {
                    print!(" ");
                }
            } else {
                if j >= i - column + 1 {
                    print!("{}", symbol);
                    symbol += 1;
                } else {
                    print!(" ");
                }
            }
        }
        println!();
    }
}
